### PR TITLE
Anna Lenksjö project mongo API

### PR DESCRIPTION
**Pull request by annalenksjo**
https://a-lenksjo-project-mongo.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,39 +1,121 @@
-import express from 'express'
-import bodyParser from 'body-parser'
+import express, { response } from 'express'
+import listEndpoints from 'express-list-endpoints'
```
Comment (ElleFrankenberg) [inline server.js:2]: Nice that you imported this to show the endpoints!

**Review by ElleFrankenberg — APPROVED** [review]
I don't have that much to comment on😊 The blue requirements are fulfilled! Endpoints and the error messages are working, the code is clean and readable with good naming👍 Really good job with this weeks project!!!

File: README.md [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+I started by setting up my account at Mongo, and connecting it to Heroku and GitHub, making sure that everything was deployed and connected as it should on Heroku and Mongo Cloud.
+After that I constructed some routes and endpoints to be able to filter and query the data. The data has two queries and 3 routes. 
+
+If I had more time I would have filtered the data more and trying to implement some more mongoose models. I would also have implemented pages with the methods skip & limit. 
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://a-lenksjo-project-mongo.herokuapp.com/
```
Comment (KarolCislowski) [inline README.md:14]: Short documentation would be nice 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,39 +1,121 @@
...
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const trackSchema = new mongoose.Schema({
+  id: Number,
```
Comment (KarolCislowski) [inline server.js:16]: Why you need to have both id and _id here??

File: server.js [hunk 4]
```diff
@@ -1,39 +1,121 @@
...
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).send({ error : 'Something went wrong'})
+  }
+})
```
Comment (KarolCislowski) [inline server.js:60]: Nice connection check 👍🏻

File: server.js [hunk 5]
```diff
@@ -1,39 +1,121 @@
...
+            $regex: new RegExp(artistName || "", "i")
+          }
+        }
+      }])
+  
+      if (tracks.length === 0) {
+        res.send('Not found')
+      } else {
+        res.json({ Length: tracks.length, Data: tracks})
+
```
Comment (KarolCislowski) [inline server.js:89]: nice filtering 👍🏻

File: server.js [hunk 6]
```diff
@@ -1,39 +1,121 @@
...
+      }
+    } catch(error) {
+        res.json(400).send({ error: 'Invalid request'})
+    }
+})
+
+app.get('/tracks/popular/mostpopular', async(req, res) => {
+
+  try {
+    const popularTracks = await Track.find({popularity:{$gte:90}})
```
Comment (KarolCislowski) [inline server.js:113]: 👍🏻

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

**Pull request closed without merging**